x/crosschain/keeper/keeper_cross_chain_tx_vote_outbound_tx.go

```diff
@@ -151,7 +151,8 @@ func (k msgServer) VoteOnObservedOutboundTx(goCtx context.Context, msg *types.Ms
 			newStatus := cctx.CctxStatus.Status.String()
 			EmitOutboundSuccess(tmpCtx, msg, oldStatus.String(), newStatus, cctx)
 		case observerTypes.BallotStatus_BallotFinalized_FailureObservation:
-			if msg.CoinType == common.CoinType_Cmd {
+			if msg.CoinType == common.CoinType_Cmd || cctx.InboundTxParams.SenderChainId == common.ZetaChain().ChainId {
```

Comment:
I can't tell the meaning of each status but just I'm curious.
1. As for `msg.CoinType == common.CoinType_Cmd`, I knew that it was a newly added type by Panruo meaning something special and concrete. What are the cases for `cctx.InboundTxParams.SenderChainId == common.ZetaChain().ChainId`? Does it overlap with the former?

2. 'CctxStatus_Aborted' leads to different subsequent workflow than 'CctxStatus_Reverted' or we are just setting a different flag to give it a different meaning for the moment?

3. Both line 146 and 178 uses set 'CctxStatus_Aborted' status. We can't unify the logic in one single flow?

Reply:
>As for msg.CoinType == common.CoinType_Cmd, I knew that it was a newly added type by Panruo meaning something special and concrete. What are the cases for cctx.InboundTxParams.SenderChainId == common.ZetaChain().ChainId? Does it overlap with the former?

They are unrelated, both are just two conditions when there is no revert and the cctx should become aborted.

>'CctxStatus_Aborted' leads to different subsequent workflow than 'CctxStatus_Reverted' or we are just setting a different flag to give it a different meaning for the moment?

We set it aborted because there is no revert tx when the sender is ZetaChain, also this scenario should not occur under the current logic.

 >Both line 146 and 178 uses set 'CctxStatus_Aborted' status. We can't unify the logic in one single flow?

Agree on that, although I think the general flow of vote_outbound_tx and vote_inbound_tx could be simplified/factorized. I would consider it for a larger refactoring PR.

Reply:
Thank you for explaining it.